LiteDB/Engine/Services/LockService.cs

```diff
@@ -54,7 +54,8 @@ public void ExitTransaction()
             // if current thread are in reserved mode, do not exit transaction (will be exit from ExitExclusive)
             if (_transaction.IsWriteLockHeld) return;
 
-            _transaction.ExitReadLock();
+            if (_transaction.IsReadLockHeld)
```

Comment:
Check before exiting, otherwise ExitReadLock throws System.Threading.SynchronizationLockException:The read lock is being released without being held